incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/config/OpenapiClientCompositeBindingAdapter.java

```diff
@@ -0,0 +1,345 @@
+import io.aklivity.zilla.runtime.binding.openapi.internal.view.PathView;
+import io.aklivity.zilla.runtime.binding.openapi.internal.view.SchemaView;
+import io.aklivity.zilla.runtime.binding.openapi.internal.view.ServerView;
+import io.aklivity.zilla.runtime.binding.tcp.config.TcpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.tls.config.TlsOptionsConfig;
+import io.aklivity.zilla.runtime.catalog.inline.config.InlineOptionsConfig;
+import io.aklivity.zilla.runtime.catalog.inline.config.InlineSchemaConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.BindingConfig;
+import io.aklivity.zilla.runtime.engine.config.BindingConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.CompositeBindingAdapterSpi;
+import io.aklivity.zilla.runtime.engine.config.ModelConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfigBuilder;
+import io.aklivity.zilla.runtime.model.core.config.IntegerModelConfig;
+import io.aklivity.zilla.runtime.model.core.config.StringModelConfig;
+import io.aklivity.zilla.runtime.model.json.config.JsonModelConfig;
+
+public final class OpenapiClientCompositeBindingAdapter implements CompositeBindingAdapterSpi
+{
+    private static final String INLINE_CATALOG_NAME = "catalog0";
+    private static final String INLINE_CATALOG_TYPE = "inline";
+    private static final String VERSION_LATEST = "latest";
+
+    private final Map<String, ModelConfig> models = Map.of(
+        "string", StringModelConfig.builder().build(),
+        "integer", IntegerModelConfig.builder().build()
+    );
+
+
+    @Override
+    public String type()
+    {
+        return OpenapiBinding.NAME;
+    }
+
+    public OpenapiClientCompositeBindingAdapter()
+    {
+    }
+
+    @Override
+    public BindingConfig adapt(
+        BindingConfig binding)
+    {
+        OpenapiOptionsConfig options = (OpenapiOptionsConfig) binding.options;
+        OpenapiConfig openapiConfig = options.openapis.get(0);
+
+        final OpenApi openApi = openapiConfig.openapi;
+        final int[] httpsPorts = resolvePortsForScheme(openApi, "https");
+        final boolean secure = httpsPorts != null;
+
+        return BindingConfig.builder(binding)
+            .composite()
+                .name(String.format(binding.qname, "$composite"))
+                .binding()
+                    .name("http_client0")
+                    .type("http")
+                    .kind(CLIENT)
+                    .inject(b -> this.injectHttpClientOptions(b, openApi))
+                    .exit(secure ? "tls_client0" : "tcp_client0")
+                    .build()
+                .inject(b -> this.injectTlsClient(b, options.tls, secure))
+                .binding()
+                    .name("tcp_client0")
+                    .type("tcp")
+                    .kind(CLIENT)
+                    .options(TcpOptionsConfig::builder)
+                        .host("") // env
+                        .ports(new int[]{0}) // env
```

Comment:
These `tcp` options should be removed.

At runtime, we need to make sure the `http` client is sending the `ProxyBeginEx` (similar to `kafka` client) to default the `tls` client and `tcp` client behavior.